File: backend/src/main/java/com/yoursunshine/backend/entity/KnobModule.java
package com.yoursunshine.backend.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Inheritance(strategy = InheritanceType.JOINED)
@Table(uniqueConstraints = {
        @UniqueConstraint(name = "PinNumberAndGroup",
                columnNames = {"pinNumber", "group_id"})
})
public abstract class KnobModule {
    @Id
    @GeneratedValue
    private Long id;

    private String title;

    private Integer pinNumber;

    @ManyToOne()
    @JoinColumn(nullable = false)
    private EspGroup group;
}

File: frontend/src/app/dtos/knobModule.ts

export interface KnobModule {
  id?: number;
  title: string;
  pinNumber: number;
  group_id?: number;
}

export interface RgbModule extends KnobModule {
  color: string;
}

export interface DimmerModule extends KnobModule {
  brightness: number;
}
